ddl/ddl_api.go

```diff
@@ -72,10 +72,81 @@ const (
 	longBlobMaxLength     = 4294967295
 )
 
-func (d *ddl) CreateSchema(ctx sessionctx.Context, schema model.CIStr, charsetInfo *ast.CharsetOpt) error {
+func (d *ddl) CreateSchema(ctx sessionctx.Context, schema model.CIStr, dbOps []*ast.DatabaseOption) error {
 	dbInfo := &model.DBInfo{Name: schema}
-	if charsetInfo != nil {
-		chs, coll, err := ResolveCharsetCollation(ast.CharsetOpt{Chs: charsetInfo.Chs, Col: charsetInfo.Col})
+	var opt *ast.CharsetOpt
+	if len(dbOps) != 0 {
+		opt = &ast.CharsetOpt{}
+		for _, val := range dbOps {
```

Comment:
This looks like it is duplicated and would benefit from a common function, similar to this:
https://github.com/pingcap/tidb/pull/27969/files#diff-b7c54f7dad7257cbe4fa04b8d76c9bed694e2e2e7c5599f45eb2172d8d4ff8a7R2289

Reply:
> similar

I will revoke the part of schema and use the common func that named SetDirectPlacementOpt to simple code after your pr be merged .

Reply:
> This looks like it is duplicated and would benefit from a common function, similar to this:
> https://github.com/pingcap/tidb/pull/27969/files#diff-b7c54f7dad7257cbe4fa04b8d76c9bed694e2e2e7c5599f45eb2172d8d4ff8a7R2289

I will revoke the part of schema and use the common func that named SetDirectPlacementOpt to simple code after your pr be merged .